contracts/erc721/UserMintableERC721.sol

```diff
@@ -430,6 +464,76 @@ contract UserMintableERC721 is Initializable, ERC721, ERC721Enumerable, ERC721Me
     );
   }
 
+  /**
+   * @dev Checks that provided document is signed by the given identity
+   * and validates and checks if the public key used is a valid
+   * SIGNING_KEY
+   * @param documentRoot bytes32 the anchored document root
+   * @param identity address Identity that signed the document
+   * @param signingRoot bytes32 The message that was signed
+   * @param singingRootProof bytes32[] proofs for signing root
+   * @param signature bytes The signature
+   * used to contract the property for precise proofs
+   * @param salt bytes32 salt for leaf construction
+   * @param proof bytes32[] proofs for leaf construction
+   */
+
+  function _requireSignedByIdentity(
+    bytes32 documentRoot,
+    address identity,
+    bytes32 signingRoot,
+    bytes32[] memory singingRootProof,
+    bytes memory signature,
+    bytes32 salt,
+    bytes32[] memory proof
+  )
+  internal
+  view
+  {
+
+    require(
+      MerkleProof.verifySha256(
+        singingRootProof,
+        documentRoot,
+        signingRoot
+      ),
+      "Signing Root not part of the document"
+    );
+
+    // Extract the public key from the signature
+    bytes32 pbKey_ = bytes32(
+      uint256(
+        signingRoot.toEthSignedMessageHash().recover(signature)
+      )
+    );
+
+    // Reconstruct the precise proof property based on the provided identity
+    // and the extracted public key
+    bytes memory property_ = abi.encodePacked(
+      hex"0300000000000001", // compact prop for "signatures_tree.signatures"
+      identity,
+      pbKey_,
+      hex"00000004" // compact prop for "signature"
+    );
+
+
+    // Check with precise proofs if the signature is part of the documentRoot
+    require(
+      MerkleProof.verifySha256(
+        proof,
+        documentRoot,
+        sha256(abi.encodePacked(property_, signature, salt))
+      ),
+      "Provided signature is not part of the document root"
+    );
+
+    // Check if the public key has a signature purpose on the provided identity
+    require(
+      _getIdentity(identity).keyHasPurpose(pbKey_, SIGNING_PURPOSE),
```

Comment:
`keyHasPurpose` is not checking if the key is revoked!

Reply:
No it does not and it will not because as I know it should return true also for revoked keys. As far as I know we are interested in the fact that identity has or had a specific key.

Reply:
@vimukthi-git did you not talk with Razvan about the key revoking?
@rdinicut it is not always the case, we identified the problem last week and started to implement a solution for it on the node.